src/rdkafka_admin.c

```diff
@@ -4801,3 +5058,1552 @@ void rd_kafka_DeleteAcls(rd_kafka_t *rk,
+            rko->rko_u.admin_result.reqtype & ~RD_KAFKA_OP_FLAGMASK;
+        rd_assert(reqtype == RD_KAFKA_OP_LISTCONSUMERGROUPS);
+
+        list_result_cnt = rd_list_cnt(&rko->rko_u.admin_result.results);
+        rd_assert(list_result_cnt == 1);
+        list_result = rd_list_elem(&rko->rko_u.admin_result.results, 0);
+        *cntp       = rd_list_cnt(&list_result->valid);
+
+        return (const rd_kafka_ConsumerGroupListing_t **)
+            list_result->valid.rl_elems;
+}
+
+const rd_kafka_error_t **rd_kafka_ListConsumerGroups_result_errors(
+    const rd_kafka_ListConsumerGroups_result_t *result,
+    size_t *cntp) {
+        int list_result_cnt, error_cnt;
+        const rd_kafka_ListConsumerGroupsResult_t *list_result;
+        const rd_kafka_op_t *rko = (const rd_kafka_op_t *)result;
+        rd_kafka_op_type_t reqtype =
+            rko->rko_u.admin_result.reqtype & ~RD_KAFKA_OP_FLAGMASK;
+        rd_assert(reqtype == RD_KAFKA_OP_LISTCONSUMERGROUPS);
+
+        list_result_cnt = rd_list_cnt(&rko->rko_u.admin_result.results);
+        rd_assert(list_result_cnt == 1);
+        list_result = rko->rko_u.admin_result.results.rl_elems[0];
+        error_cnt   = rd_list_cnt(&list_result->errors);
+        if (error_cnt == 0) {
+                *cntp = 0;
+                return NULL;
+        }
+        *cntp = error_cnt;
+        return (const rd_kafka_error_t **)list_result->errors.rl_elems;
+}
+
+/**@}*/
+
+/**
+ * @name Describe consumer groups
+ * @{
+ *
+ *
+ *
+ *
+ */
+
+/**
+ * @brief Create a new MemberDescription object. This object is used for
+ *        creating a ConsumerGroupDescription.
+ *
+ * @param client_id The client id.
+ * @param consumer_id The consumer id (or member id).
+ * @param group_instance_id (optional) The group instance id
+ *                          for static membership.
+ * @param host The consumer host.
+ * @param assignment A member assignment containing a list of topic-partitions.
+ * @return A new allocated MemberDescription object.
+ *         Use rd_kafka_MemberDescription_destroy() to free when done.
+ */
+static rd_kafka_MemberDescription_t *
+rd_kafka_MemberDescription_new(const char *client_id,
+                               const char *consumer_id,
+                               const char *group_instance_id,
+                               const char *host,
+                               rd_kafka_MemberAssignment_t assignment) {
+        rd_kafka_MemberDescription_t *member;
+        member              = rd_calloc(1, sizeof(*member));
+        member->client_id   = rd_strdup(client_id);
+        member->consumer_id = rd_strdup(consumer_id);
+        if (group_instance_id)
+                member->group_instance_id = rd_strdup(group_instance_id);
+        member->host       = rd_strdup(host);
+        member->assignment = assignment;
+        member->assignment.partitions =
+            rd_kafka_topic_partition_list_copy(member->assignment.partitions);
```

Comment:
Two things regarding this line:

1. We're currently copying member->assignment.partitions into itself. Rather, it should be:
```
member->assignment.partitions =
            rd_kafka_topic_partition_list_copy(assignment.partitions);
```

2. We should have a NULL check somewhere on assignment.partitions, since `rd_kafka_topic_partition_list_copy` assumes the argument passed to it will be not NULL. But assignment.partitions can indeed be NULL. 
How we will handle the NULL value is something I can't confirm, we can either do member->assignment.partitions = NULL or else member->assignment.partitions = empty partition list

(2) sometimes leads to a segfault, for which I'm attaching a partial bt.

```
#0  rd_kafka_topic_partition_list_copy (src=src@entry=0x0) at rdkafka_partition.c:2755
#1  0x00007ffff7e77b3f in rd_kafka_MemberDescription_new (
    client_id=0x7fffa4005db0 "test.client.1", 
    consumer_id=0x7fffa4004b80 "test.client.1-641dfe50-6d1f-4b84-8ff2-ad84094b86dd", 
    group_instance_id=0x0, host=0x7fffa40051c0 "/172.19.0.1", assignment=...)
    at rdkafka_admin.c:5956
#2  0x00007ffff7e549a5 in rd_list_copy_to (dst=dst@entry=0x7fffa40051f0, 
    src=src@entry=0x7fffc15d66a0, 
    copy_cb=copy_cb@entry=0x7ffff7e77b90 <rd_kafka_MemberDescription_list_copy>, 
    opaque=opaque@entry=0x0) at rdlist.c:457
#3  0x00007ffff7e77d9b in rd_kafka_ConsumerGroupDescription_new (
    group_id=group_id@entry=0x7fffa8001e30 "testgroup-108073486942453323", 
    is_simple_consumer_group=is_simple_consumer_group@entry=0 '\000', 
    members=members@entry=0x7fffc15d66a0, 
    partition_assignor=partition_assignor@entry=0x7fffa40035f0 "range", 
    state=RD_KAFKA_CONSUMER_GROUP_STATE_STABLE, coordinator=coordinator@entry=0x7fffa4005190, 
    error=0x0) at rdkafka_admin.c:6072
#4  0x00007ffff7e83a77 in rd_kafka_DescribeConsumerGroupsResponse_parse (rko_req=<optimized out>, 
    rko_resultp=0x7fffc15d6960, reply=0x7fffac000f00, errstr=0x7fffc15d6970 "h\177]\301\377\177", 
    errstr_size=512) at rdkafka_admin.c:6446
```

Reply:
Good find, Milind :+1:
I have a fix on the way

Reply:
https://github.com/confluentinc/librdkafka/pull/3995/commits/ec016519d4623cfee566cb951e2fa1f8062929d8